instrumentation/netty/netty-4-common/javaagent/src/main/java/io/opentelemetry/javaagent/instrumentation/netty/common/AbstractNettyChannelPipelineInstrumentation.java

```diff
@@ -125,6 +149,49 @@ public static void removeHandler(
       if (ourHandler != null) {
         pipeline.remove(ourHandler);
         contextStore.put(handler, null);
+      } else if (handler
+          .getClass()
+          .getName()
+          .startsWith("io.opentelemetry.javaagent.instrumentation.netty.")) {
+        pipeline.removeLast();
+      }
+    }
+  }
+
+  @SuppressWarnings("unused")
+  public static class AddAfterAdvice {
+
+    @Advice.OnMethodExit(suppress = Throwable.class)
+    public static void addAfterHandler(
+        @Advice.This ChannelPipeline pipeline,
+        @Advice.Argument(value = 0, readOnly = false) String name) {
+      ChannelHandler handler = pipeline.get(name);
+      if (handler != null) {
+        ContextStore<ChannelHandler, ChannelHandler> contextStore =
+            InstrumentationContext.get(ChannelHandler.class, ChannelHandler.class);
+        ChannelHandler ourHandler = contextStore.get(handler);
+        if (ourHandler != null) {
+          name = ourHandler.getClass().getName();
+        }
+      }
+    }
+  }
+
+  @SuppressWarnings("unused")
+  public static class AddAfterWithGroupAdvice {
+
+    @Advice.OnMethodExit(suppress = Throwable.class)
+    public static void addAfterHandler(
+        @Advice.This ChannelPipeline pipeline,
+        @Advice.Argument(value = 1, readOnly = false) String name) {
```

Comment:
Can you add `takesArgument(1, String.class)` to the method matcher? In general we aim to enclose all types that we use in advice classes (arguments, return types) in the method matcher.

Reply:
yes, of course